h2o-algos/src/main/java/hex/tree/SharedTree.java

```diff
@@ -165,6 +165,11 @@ protected void ignoreInvalidColumns(int npredictors, boolean expensive) {
 
     PlattScalingHelper.initCalibration(this, _parms, expensive);
   }
+  
+  
+  protected void validateMaxDepth(){
+    if (_parms._max_depth <= 0) error ("_max_depth", "_max_depth must be > 0.");
```

Comment:
why chosing this default implementation for `validateMaxDepth` here if you override it with the same logic in all subclasses?

Reply:
I am with seb on this. also the override is the same in all subclasses, so ti does not make much sense.
also  agree that forcing 0 to mean unlimited is uneccessary I think <=0 is better for the user

Reply:
Yes, the override is in all classes and can be replaced.

Regarding the convention - we kind-of agreed with @sebhrusen (also what @michalkurka says) that 0 means disabled and -1 is automatic behavior. In this case, disabling max_depth restriction implies making the trees as deep as technically possible. Therefore 0. Still valid @sebhrusen ? See api-updates channel :)

Reply:
Yes, I'm ok with the distinction `disabled` (=0) vs `auto` (<0).
My original argument was that for some use-cases, the distinction is not always clear, although here, we can imagine a case were "auto" would mean "use an estimated value". TBH, it's all theoretical because such a feature doesn't exist, but it shows that we may want to use negative values for a different semantic in the future.